src/dotnet/APIView/APIViewWeb/Models/RenderedCodeFile.cs

```diff
@@ -43,15 +50,83 @@ internal CodeLine[] RenderText(bool skipDiff = false)
         {
             if (skipDiff)
             {
-                return CodeFileRenderer.Instance.Render(CodeFile, enableSkipDiff: skipDiff);
+                RenderResult = CodeFileRenderer.Instance.Render(CodeFile, enableSkipDiff: skipDiff);
+                return RenderResult.CodeLines;
             }
 
             if (_renderedText == null)
             {
-                _renderedText = CodeFileRenderer.Instance.Render(CodeFile);
+                RenderResult = CodeFileRenderer.Instance.Render(CodeFile);
+                _renderedText = RenderResult.CodeLines;
             }
 
             return _renderedText;
         }
+
+        public CodeLine[] GetCodeLineSection(int sectionId)
+        {
+            var section = RenderResult.Sections[sectionId];
```

Comment:
Shouldn't we check to ensure Sections size > sectionId to avoid any index out of array? I assume currently that's not an issue by handling it in the caller side? But it's good to add this check to avoid an issue in the future if someone accidentally refer this API without for a review without any sections.